### ISSUE TITLE
blocks_in_conditions can't deal with return expression in condition

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#9911]
### Summary

.

### Lint Name

blocks_in_if_conditions

### Reproducer

I tried this code:

```rust
fn main() {
    if { return } {}
}

```

I saw this happen:
`cargo clippy --fix -- -Aclippy::all -Wclippy::blocks_in_if_conditions`
```
    Checking clpy v0.1.0 (/tmp/clpy)
warning: failed to automatically apply fixes suggested by rustc to crate `clpy`

after fixes were automatically applied the compiler reported errors within these files:

  * src/main.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error: expected `{`, found `}`
 --> src/main.rs:3:1
  |
3 | }
  | ^ expected `{`
  |
note: the `if` expression is missing a block after this condition
 --> src/main.rs:2:8
  |
2 |     if return {}
  |        ^^^^^^^^^

error: aborting due to previous error

Original diagnostics will follow.

warning: unreachable block in `if` or `while` expression
 --> src/main.rs:2:19
  |
2 |     if { return } {}
  |          ------   ^^ unreachable block in `if` or `while` expression
  |          |
  |          any code following this expression is unreachable
  |
  = note: `#[warn(unreachable_code)]` on by default

warning: omit braces around single expression condition
 --> src/main.rs:2:8
  |
2 |     if { return } {}
  |        ^^^^^^^^^^ help: try: `return`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#blocks_in_if_conditions
  = note: requested on the command line with `-W clippy::blocks-in-if-conditions`

warning: `clpy` (bin "clpy") generated 2 warnings (run `cargo fix --bin "clpy"` to apply 1 suggestion)
warning: `clpy` (bin "clpy" test) generated 2 warnings (2 duplicates)
    Finished dev [unoptimized + debuginfo] target(s) in 0.34s
```

### Version

```text
rustc 1.67.0-nightly (c5d82ed7a 2022-11-19)
binary: rustc
commit-hash: c5d82ed7a4ad94a538bb87e5016e7d5ce0bd434b
commit-date: 2022-11-19
host: x86_64-unknown-linux-gnu
release: 1.67.0-nightly
LLVM version: 15.0.4
```
